llvm/lib/Transforms/Utils/SCCPSolver.cpp

```diff
@@ -157,6 +162,15 @@ static bool refineInstruction(SCCPSolver &Solver,
                           GEPNoWrapFlags::noUnsignedWrap());
       Changed = true;
     }
+  } else if (match(&Inst, m_And(m_Value(X), m_LowBitMask(RHSC)))) {
```

Comment:
Could try generalizing this away from LowBitMask in a followup. For CVP this reduces compile-time, but for IPSCCP it shouldn't matter.